### PR TITLE
feat: habit participant activity, reordering habits list, dropdown menus

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added participant activity display, dropdown menus (native menus using Zeego) + reordering habits using mmkv, and improved some caching logic and types.

Unfortunately, when you click the menu 3 dots button when it is on the bottom half of the screen, it opens the menu upwards and reverses the direction of all the menu items. This puts the "move up in list" button below the "move down in list" button. This is just an iOS thing. I think I'll just ignore it, it's not that bad.

Right now, the edit & delete habit buttons do nothing

https://github.com/user-attachments/assets/25cbd2db-afe0-4278-96c7-f13f0ac2b7f9

### What changed?

- Added dropdown menu to reorder habits in the list
- Enhanced habit cards with participant activity indicators
- Added translucent color utility for the badge color
- Integrated zeego library for dropdown menus
- Added participant activity badges showing who's active today
- Improved profile picture display with active/inactive status indicators
- Restructured habit data types for better type safety

### How to test?

1. Open the habits list screen
2. Click the ellipsis menu on any habit card
3. Try moving habits up/down in the list
4. Verify participant pictures show correct active/inactive states
5. Check that the "X active today" badge updates correctly
6. Confirm habit order persists after app reload

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/141?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#141** <a href="https://app.graphite.dev/github/pr/habits-together/app/141?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/141?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#140** <a href="https://app.graphite.dev/github/pr/habits-together/app/140?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/core/hooks/use-selected-theme.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const useSelectedTheme = () => {
 export const loadSelectedTheme = () => {
   const theme = storage.getString(SELECTED_THEME);
   if (theme !== undefined) {
-    console.log('theme', theme);
```
Comment (AnkushSarkar10) [inline src/core/hooks/use-selected-theme.tsx:35]: remove

File: src/core/get-translucent-color.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// you can define a number as rgba using #RRGGBBAA
+// so we just append the opacity value to the color hex value
+export function getTranslucentColor(colorHexValue: string, opacity: number) {
```
Comment (AnkushSarkar10) [inline src/core/get-translucent-color.tsx:3]: mention opacity bw 0-1

**Review by AnkushSarkar10 — CHANGES_REQUESTED** [review]
bitch ass

**Review by AnkushSarkar10 — APPROVED** [review]
